models/demos/utils/llm_demo_utils.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from loguru import logger
+from models.perf.benchmarking_utils import BenchmarkProfiler, BenchmarkData
+
+
+def create_benchmark_data(profiler: BenchmarkProfiler, measurements: dict, N_warmup_iter: dict, targets: dict):
+    """
+    Create a benchmark data object and populate the object with the given measurements.
+
+    Pre-requisites:
+    - The measurements dictionary should contain the following keys: "compile_prefill", "compile_decode", "prefill_t/s", "decode_t/s", "decode_t/s/u"
+    - The profiler object should contain the start and end times for the steps "compile_prefill", "compile_decode", "inference_prefill", "inference_decode"
+
+    Optional (should be provided if measuring perf, not required for token verification):
+    - The targets dictionary should contain the following keys: "prefill_t/s", "decode_t/s", "decode_t/s/u"
+    - The N_warmup_iter dictionary should contain the following keys: "inference_prefill", "inference_decode"
+
+    Optional (should be provided if doing token verification, not required for perf):
+    - The measurements and targets dictionaries should contain the key "token_verification"
+    """
+
+    assert all(
```

Comment:
do we need to relax this constraint? What if we have tests with no prefill, no decode? Will they have to pass in values for non-existing measurements?

Reply:
This function is only intended for llm demos which all have prefill/decode measurements. Any other non-demo tests can call BenchmarkData::add_measurement in its own script or utility.